aviatrix/resource_aviatrix_device_transit_gateway_attachment.go

```diff
@@ -187,13 +198,19 @@ func resourceAviatrixDeviceTransitGatewayAttachmentCreate(d *schema.ResourceData
 
 	manualBGPCidrs := getStringSet(d, "manual_bgp_advertised_cidrs")
 	if len(manualBGPCidrs) > 0 {
-		err = client.EditTransitConnectionBGPManualAdvertiseCIDRs(attachment.TransitGatewayName, attachment.ConnectionName, manualBGPCidrs)
+		err := client.EditTransitConnectionBGPManualAdvertiseCIDRs(attachment.TransitGatewayName, attachment.ConnectionName, manualBGPCidrs)
 		if err != nil {
 			return fmt.Errorf("could not edit manual advertised cidrs: %v", err)
 		}
 	}
 
-	return err
+	if d.Get("enable_event_triggered_ha").(bool) {
+		if err := client.EnableSite2CloudEventTriggeredHA(d.Get("vpc_id").(string), attachment.ConnectionName); err != nil {
```

Comment:
Is the vpc_id set by an api call before this point?

Reply:
Yes, see Line 189 I call the Read function